New Pattern: AWS Lambda REST API behind Amazon ALB with resource-path Listener Rule (Python & SAM) (#2027)
aws-samples/serverless-patterns: alb-lambda-rest-api-sam-py/README.md

# AWS Lambda RESTful API with Amazon ALB and Path Based Listener Rules

> This configuration employs Python 3.9 and the Serverless Application Model (SAM) CLI to establish an Application Load
> Balancer with path-based listener rules, paired with an AWS Lambda RESTful API function as the target.

Reviewer: ```suggestion
This configuration pattern creates an Application Load
Balancer with path-based listener rules, paired with an AWS Lambda RESTful API function as the target. It uses  Python 3.9 and the Serverless Application Model (SAM) CLI.
```

  have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls
  and manage AWS resources.
* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (
  AWS SAM) installed
* [Python 3 installed](https://www.python.org/downloads/)
* [Docker](https://www.docker.com/products/docker-desktop/)

## Deployment Instructions

1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
    ``` 
    git clone https://github.com/aws-samples/serverless-patterns
    ```
2. Change directory to the pattern directory:
    ```
    cd alb-lambda-rest-api-sam-py
    ```
3. Install dependencies and build:
    ```
    pipenv install
    ```
4. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
    ```
    sam validate && sam build && sam deploy --guided
    ```

> 5. During the prompts:

Reviewer: When I run `sam deploy --guided` the  stack also asks me for a VPC id and subnets. Please include instructions what the user needs to do here and what format is expected.

* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already
  have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls
  and manage AWS resources.
* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (
  AWS SAM) installed
* [Python 3 installed](https://www.python.org/downloads/)
* [AWS Lambda Powertools for Python](https://docs.powertools.aws.dev/lambda/python/latest/)
* [Docker](https://www.docker.com/products/docker-desktop/)

## Deployment Instructions

1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
    ``` 
    git clone https://github.com/aws-samples/serverless-patterns
    ```
2. Change directory to the pattern directory:
    ```
    cd alb-lambda-rest-api-sam-py
    ```
3. Install dependencies and build:
    ```

>     pipenv install

Reviewer: If I do not run `pipenv requirements > requirements.txt` before `sam deploy` the Powertools dependency is not installed. Is the command missing here or did I miss something?

Author: I had this step in Testing the example locally. But it makes sense to have the instruction here as well.
